frontend/lib/routes.dart

```diff
@@ -42,64 +41,201 @@ class AppGoRouter {
-          }
-          return BlogList();
-        }
-      ),
+          path: Routes.blogView,
+          pageBuilder: (BuildContext context, GoRouterState state) {
+            String? slug = state.params["slug"];
+
+            if (slug != null) {
+              return CustomTransitionPage(
+                  child: BlogView(slug: slug),
+                  transitionsBuilder:
+                      (context, animation, secondaryAnimation, child) {
+                    return FadeTransition(
+                      opacity: CurveTween(curve: Curves.easeInOutCirc)
+                          .animate(animation),
+                      child: child,
+                    );
+                  });
+            }
+            return CustomTransitionPage(
+                child: BlogList(),
+                transitionsBuilder:
+                    (context, animation, secondaryAnimation, child) {
+                  return FadeTransition(
+                    opacity: CurveTween(curve: Curves.easeInOutCirc)
+                        .animate(animation),
+                    child: child,
+                  );
+                });
+          }),
       GoRoute(
-        path: Routes.design,
-        builder: (BuildContext context, GoRouterState state) => Design(),
-      ),
+          path: Routes.design,
+          pageBuilder: (BuildContext context, GoRouterState state) {
+            return CustomTransitionPage(
+                child: Design(),
+                transitionsBuilder:
+                    (context, animation, secondaryAnimation, child) {
+                  return FadeTransition(
+                    opacity: CurveTween(curve: Curves.easeInOutCirc)
+                        .animate(animation),
+                    child: child,
+                  );
+                });
+          }),
       GoRoute(
-        path: Routes.lendLibrary,
-        builder: (BuildContext context, GoRouterState state) => LendLibrary(),
-      ),
+          path: Routes.lendLibrary,
+          pageBuilder: (BuildContext context, GoRouterState state) {
+            return CustomTransitionPage(
+                child: LendLibrary(),
+                transitionsBuilder:
+                    (context, animation, secondaryAnimation, child) {
+                  return FadeTransition(
+                    opacity: CurveTween(curve: Curves.easeInOutCirc)
+                        .animate(animation),
+                    child: child,
+                  );
+                });
+          }),
       GoRoute(
-        path: Routes.team,
-        builder: (BuildContext context, GoRouterState state) => Team(),
-      ),
+          path: Routes.team,
+          pageBuilder: (BuildContext context, GoRouterState state) {
+            return CustomTransitionPage(
+                child: Team(),
+                transitionsBuilder:
+                    (context, animation, secondaryAnimation, child) {
+                  return FadeTransition(
```

Comment:
Oh wow, is there not a setting to apply a default to all of them? If it requires this much extra code, maybe it's not worth it!